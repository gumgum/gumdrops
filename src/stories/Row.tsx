import React, { ReactChild, ReactChildren } from 'react';
import cx from 'classnames';

export interface RowProps {
    children: ReactChildren | ReactChild;
    className?: string;
}

export const Row: React.FC<RowProps> = ({ className, children, ...otherProps }) => (
    <div className={cx('gds-layout__row', className)} {...otherProps}>
        {children}
    </div>
);
