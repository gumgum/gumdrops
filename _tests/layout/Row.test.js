import renderer from 'react-test-renderer';
import React from 'react';
import Row from '../../components/layout/Row';

const props = {
    className: 'cool-stuff'
};

test('Expect <Row> to render properly', () => {
    const tree = renderer
        .create(
            <Row {...props}>
                <div>Content</div>
            </Row>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
