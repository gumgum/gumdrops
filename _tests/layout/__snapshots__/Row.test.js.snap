// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Row> to render properly 1`] = `
<div
  className="gds-layout__row cool-stuff"
>
  <div>
    Content
  </div>
</div>
`;
